scripts/color_districts.py

```diff
@@ -6,41 +6,62 @@
     texas_districts = geojson.load(texas_districts_file)
     texas_districts_file.close()
 
-districts = []
+districts = list()
 
+# Adds districts to list (as Shapely objects)
 for feature in texas_districts["features"]:
     districts.append(shape(feature["geometry"]))
 
 adjacencies = [set() for i in range(len(districts))]
 
+# Creates adjacency list for each district
 for i in range(len(districts)):
     for j in range(i + 1, len(districts)):
         if type(districts[i].intersection(districts[j])) is LineString or type(districts[i].intersection(districts[j])) is MultiLineString:
             adjacencies[i].add(j)
             adjacencies[j].add(i)
 
-coloring = [-1 for i in range(len(districts))]
+degree_list = [list()] * len(districts)
+coloring_order = list()
+coloring = [-1] * len(districts)
 
-# Assign the first vertex with the lowest number.
-coloring[0] = 0
+# Creates list of districts with degree i (for i between 0 and number of districts - 1)
+for i in range(len(districts)):
+    degree_list[len(adjacencies[i])].append(i)
+
+# Establishes order in which districts should be colored to ensure 6-coloring
+for i in range(len(districts)):
+    minimum_degree = next(index for (index, lst) in enumerate(degree_list) if lst)
+    minimum_vertex = degree_list[minimum_degree].pop(0)
+    coloring_order.append(minimum_vertex)
 
-for vertex in range(1, len(coloring)):
+    for neighbor in adjacencies[minimum_vertex]:
+        for index in range(len(degree_list)):
+            if neighbor in degree_list[index]:
+                degree_list[index].remove(neighbor)
+                degree_list[index - 1].append(neighbor)
+                break
+
+# Assign the first district with the lowest number.
+coloring[coloring_order.pop(0)] = 0
```

Comment:
There's a couple of points of improvement possible here. One you should be able to combine this and the next loop into a neater package. I also suspect there's a better way to directly leverage the coloring order into color rather than have two complex intermediary steps. However, you're still new at this so I wouldn't worry to much, but sometimes once you've written code it can help to go back through and rewrite it once you understand how the implementation has to go.